pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -66,11 +66,12 @@ const (
 )
 
 type RouterConfig struct {
-	Image         string `json:"image"`
-	CpuRequest    string `json:"cpuRequest"`
-	CpuLimit      string `json:"cpuLimit"`
-	MemoryRequest string `json:"memoryRequest"`
-	MemoryLimit   string `json:"memoryLimit"`
+	Image         string              `json:"image"`
+	CpuRequest    string              `json:"cpuRequest"`
+	CpuLimit      string              `json:"cpuLimit"`
+	MemoryRequest string              `json:"memoryRequest"`
+	MemoryLimit   string              `json:"memoryLimit"`
+	Headers       map[string][]string `json:"headers"`
```

Comment:
Why it is a map of strings ?

Reply:
Today, I added the feature so that user can specify which headers to propagate.
Now, if in future someone wants to add section like "remove" to remove certain headers or "transform" to update certain headers before passing down to nodes then this design makes things extensible.

Reply:
I tried to learn from how [others](https://www.apollographql.com/docs/router/configuration/header-propagation/) are doing it and made it even simpler to start with.

Reply:
Can we add some comments here ？

Reply:
Added.